internal/configs/version1/nginx-plus.ingress.tmpl

```diff
@@ -73,7 +73,28 @@ server {
 	{{range $AppProtectLogConf := $server.AppProtectLogConfs}}app_protect_security_log {{$AppProtectLogConf}};
 	{{end}}
 	{{- end}}
-	
+
+	{{- if $server.AppProtectDosEnable}}
+	app_protect_dos_enable {{$server.AppProtectDosEnable}};
+	{{if $server.AppProtectDosPolicy}}app_protect_dos_policy_file {{$server.AppProtectDosPolicy}};{{end}}
+	{{- end}}
+	{{- if $server.AppProtectDosLogEnable}}
+	app_protect_dos_security_log_enable {{$server.AppProtectDosLogEnable}};
+	{{if $server.AppProtectDosLogConf}}app_protect_dos_security_log {{$server.AppProtectDosLogConf}};{{end}}
+	    {{- if $server.AppProtectDosAccessLogDst}}
+        set $loggable '0';
+        access_log syslog:server={{ .AppProtectDosAccessLogDst }} log_dos if=$loggable;
+        {{- end}}
+	{{- end}}
+
+	{{- if $server.AppProtectDosMonitor}}
+	app_protect_dos_monitor "{{$server.AppProtectDosMonitor}}";
+	{{- end}}
+
+    {{- if $server.AppProtectDosName}}
+    app_protect_dos_name "{{$server.AppProtectDosName}}";
```

Comment:
What will happen if the configured name isn't unique? Let's say there is two Ingress resources and they both configure the same DOS name.

Reply:
Lets look at this example:

server {​​​​​​​
   server_name coffee.com;
   listen 80;
   app_protect_dos_monitor coffee.com/abc;
   app_protect_dos_name example;
   location /abc {​​​​​​​
        app_protect_dos_name example;
        proxy_pass app1;
   }​​​​​​​


   location / {​​​​​​​
        app_protect_dos_name example;
        proxy_pass app1;
   }​​​​​​​
}​​​​​​​

We see that there is same name to different locations.
What will happen is that the admd will handle to request from both / and /abc and like one napdos service.
So, if there will be mitigation on the napdos service, both the location perform the mitigation.

Reply:
I see a potential problem here.

Let's say we have two Ingress resources:

first one, managed by team A:

```
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: team-a-ingress
  annotations:
    kubernetes.io/ingress.class: "nginx"
    appprotectdos.f5.com/app-protect-dos-enable: "True"
    . . .
    appprotectdos.f5.com/app-protect-dos-name: "teamA.example.com"
spec:
  rules:
    - host: teamA.example.com 
```   

second one, managed by team B:

```
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: team-a-ingress
  annotations:
    kubernetes.io/ingress.class: "nginx"
    appprotectdos.f5.com/app-protect-dos-enable: "True"
    . . .
    appprotectdos.f5.com/app-protect-dos-name: "teamA.example.com"
spec:
  rules:
    - host: teamB.example.com 
```

as you can see, team B can put `teamA` into the name annotation. This mean that team B can affect DOS configuration of team A, because admd will treat them as one service now. 

I wonder if we should allow that?